### PR TITLE
feat: add product name tags

### PR DESCRIPTION
**Pull request by TimDiam0nd**
Added tags for internet explorer and node js related rules. 
Tags have been chosen based off the value in `msg` i.e. the internet explorer tags have been added to rules that contain "IE" in the `msg` field

### CODE REVIEW

**Comment by airween** [comment]
Hi @TimDiam0nd,

thanks for the PR.

First of all, could you take a look at [this lintian](https://github.com/coreruleset/coreruleset/actions/runs/10761963621/job/29841733587?pr=3815#step:9:62) error? Seems like the `platform-platform-internet-explorer` is missing from the list.

(But perhaps this is a typo, I'm not sure you want to add a tag with name `platform-platform` :smiley:)

**Comment by TimDiam0nd** [comment]
Hey, apologies about that, that should be fixed now (i initially wasnt aware of the approved tags list). Happy to rename the nodejs tag to something else if you would like

**Comment by airween** [comment]
> Hey, apologies about that, that should be fixed now (i initially wasnt aware of the approved tags list). Happy to rename the nodejs tag to something else if you would like

No worries, thanks for contributing.

`platform-node-js` suits me, but you've added another tag (for eg.) [here](https://github.com/coreruleset/coreruleset/pull/3815/files#diff-49c831b25b258f29245a1fa974e05ccdffd3a16e2294bab352750ce88056e2bfR549). All used tags must listen in file `APPROVED_TAGS`.

But what would be good to explain in the [PR's summary](https://github.com/coreruleset/coreruleset/pull/3815#issue-2512491012) is how did you collect the rules what need to have these tags.

**Comment by airween** [comment]
> but you've added another tag (for eg.) [here](https://github.com/coreruleset/coreruleset/pull/3815/files#diff-49c831b25b258f29245a1fa974e05ccdffd3a16e2294bab352750ce88056e2bfR549).

Sorry, now I see that tag had already added.

**Comment by TimDiam0nd** [comment]
> But what would be good to explain in the [PR's summary](https://github.com/coreruleset/coreruleset/pull/3815#issue-2512491012) is how did you collect the rules what need to have these tags.
PR Summary updated as per above

File: rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf [hunk 1]
```diff
@@ -28,7 +28,7 @@ SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 1" "id:934012,phase:2,pass,nolog,tag:'O
 # - funcster: __js_function
 #
 # See:
-# https://opsecx.com/index.php/2017/02/08/exploiting-node-js-deserialization-bug-for-remote-code-execution/
+# https://opsecx.com/index.php/2017/02/08/exploiting-platform-node-js-deserialization-bug-for-remote-code-execution/
```
Comment (fzipi) [inline rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf:31]: This change looks that was part of an automated change. Is the url working?

File: rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf [hunk 2]
```diff
@@ -703,7 +716,7 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS|REQU
 # Unsafe JS functions like eval(), setInterval(), setTimeout()
 # Unsafe JS constructor new Function()
 # https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#dangerous-contexts
-# https://snyk.io/blog/5-ways-to-prevent-code-injection-in-javascript-and-node-js/
+# https://snyk.io/blog/5-ways-to-prevent-code-injection-in-javascript-and-platform-node-js/
```
Comment (fzipi) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf:719]: Same here.

**Review by fzipi — CHANGES_REQUESTED** [review]
Isn't `nodejs` instead of `node-js`?

File: rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf (hunk 1 above)
Comment (TimDiam0nd) [reply rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf]: Apoligies, fixed that

Comment (fzipi) [reply rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf]: Maybe it's me, but I don't see the fix... 🤷

Comment (RedXanadu) [reply rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf]: Getting a _404 Not Found_ here, for that link..

Comment (TimDiam0nd) [reply rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf]: yep really sorry about that, ive finally reverted it all and made sure i havent change the urls in the last commit

**Comment by RedXanadu** [comment]
Open questions from today's issues meeting:

- The references to "IE XSS filters": is it just that those XSS-matching regular expressions came from Microsoft? And they put them into IE 8 many, many years ago?
- What do we mean by a 'platform' tag? The target of the attack we are detecting? The source of the knowledge to write the rule? The user agent for a would-be attack/vulnerability?

**Comment by RedXanadu** [comment]
Re-opening as this is unresolved and we decided to investigate it further.

**Pull request closed without merging**